### PR TITLE
Merge 1.6 -> 1.7

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/CompletenessRepository.php [hunk 1]
```diff
@@ -63,22 +74,33 @@ public function getProductsCountPerChannels($localeCode)
...
+                WHERE p.is_enabled = 1
+                AND t.locale = '%locale%'
+            ) as p 
+            JOIN pim_catalog_channel ch on ch.id = p.channel_id
+            JOIN pim_catalog_channel_locale cl ON cl.channel_id = ch.id
+            JOIN pim_catalog_locale lo ON lo.id = cl.locale_id
+            LEFT JOIN pim_catalog_completeness co
+            ON co.locale_id = lo.id AND co.channel_id = ch.id AND co.product_id = p.id AND co.ratio = 100
+            GROUP BY ch.id, lo.id, ch.label, lo.code 
+        ) as co;
```
Comment (ahocquard) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/CompletenessRepository.php]:
```
       SELECT co.label, co.code as locale, co.total FROM (
            SELECT t.foreign_key as channel_id, lo.id as locale_id, t.label, lo.code, COUNT(co.product_id) as total 
            FROM 
            (
                SELECT DISTINCT ch.id as channel_id, p.id FROM pim_catalog_channel ch
                JOIN %category_table% ca ON ca.root = ch.category_id
                JOIN %category_join_table% cp ON cp.category_id = ca.id
                JOIN %product_table% p ON p.id = cp.product_id
                WHERE p.is_enabled = 1
            ) as p 
            JOIN pim_catalog_channel_translation t on t.foreign_key = p.channel_id
            JOIN pim_catalog_channel_locale cl ON cl.channel_id = p.channel_id
            JOIN pim_catalog_locale lo ON lo.id = cl.locale_id
            LEFT JOIN pim_catalog_completeness co
            ON co.locale_id = lo.id AND co.channel_id = t.foreign_key AND co.product_id = p.id AND co.ratio = 100
            WHERE t.locale = 'en_US'
            GROUP BY t.foreign_key, lo.id, t.label, lo.code 
        ) as co;
```

**Comment by ahocquard** [comment]
I've reworked an tested the request, it should work better :)

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/CompletenessRepository.php [hunk 2]
```diff
@@ -63,22 +73,32 @@ public function getProductsCountPerChannels($localeCode)
...
+                JOIN %category_join_table% cp ON cp.category_id = ca.id
+                JOIN %product_table% p ON p.id = cp.product_id
+                WHERE p.is_enabled = 1
+            ) as p 
+            JOIN pim_catalog_channel_translation t on t.foreign_key = p.channel_id
+            JOIN pim_catalog_channel_locale cl ON cl.channel_id = p.channel_id
+            JOIN pim_catalog_locale lo ON lo.id = cl.locale_id
+            LEFT JOIN pim_catalog_completeness co
+            ON co.locale_id = lo.id AND co.channel_id = t.foreign_key AND co.product_id = p.id AND co.ratio = 100
+            WHERE t.locale = 'en_US'
```
Comment (ahocquard) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/CompletenessRepository.php]: Sorry, I forgot to replace it by '%locale%'

Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/CompletenessRepository.php]: Hem, wouldn't that mean the problem is also in 1.5?

**Pull request merged**